torch/quantization/ns/graph_passes.py

```diff
@@ -299,7 +304,8 @@ def load_arg(a):
             # cast dtype from the dtype of node_c's input to the dtype of
             # node_a's input (dequant, etc)
             dtype_cast_node = _insert_dtype_cast_after_node(
-                node_a, node_c, node_c.args[0], gm_a, gm_b, node_b.name + '_dtype_cast_')
+                node_a, node_c, node_c.args[0], gm_a, gm_b, graph_c,
```

Comment:
Just to be sure: graph_c shares weights with graph a and b, so a user cannot mutate a and b and assume that c is unchanged,

Reply:
More specifically,
1. graph_c and graph_b both have gm_b as the parent module (a graph itself has no weights, it points to attributes of the parent module)
2. the shadowing logic adds references from gm_b to the necessary objects (modules, packed_params, attrs) of gm_a